cdap-ui/app/cdap/components/DataPrep/index.js

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+// import DataPrepTopPanel from 'components/DataPrep/TopPanel';
+// import DataPrepTable from 'components/DataPrep/DataPrepTable';
+// import DataPrepSidePanel from 'components/DataPrep/DataPrepSidePanel';
+// import DataPrepCLI from 'components/DataPrep/DataPrepCLI';
+import MyDataPrepApi from 'api/dataprep';
+import cookie from 'react-cookie';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import DataPrepServiceControl from 'components/DataPrep/DataPrepServiceControl';
+import ee from 'event-emitter';
+import NamespaceStore from 'services/NamespaceStore';
+
+/**
+ *  Data Prep requires a container component (DataPrepHome) that will handle routing within React.
+ *  This is beacause DataPrep component will be included in Pipelines,
+ **/
+export default class DataPrep extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      backendDown: false
+    };
+
+    this.toggleBackendDown = this.toggleBackendDown.bind(this);
+    this.eventEmitter = ee(ee);
+
+    this.eventEmitter.on('DATAPREP_BACKEND_DOWN', this.toggleBackendDown);
+
+  }
+
+  componentWillMount() {
+    let workspaceId = cookie.load('DATAPREP_WORKSPACE');
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    let params = {
+      namespace,
+      workspaceId: workspaceId,
+      limit: 100
+    };
+
+    MyDataPrepApi.execute(params)
+      .subscribe((res) => {
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setWorkspace,
+          payload: {
+            data: res.value,
+            headers: res.header,
+            workspaceId
+          }
+        });
+      }, (err) => {
+        if (err.statusCode === 503) {
+          console.log('backend not started');
+          this.eventEmitter.emit('DATAPREP_BACKEND_DOWN');
+          return;
+        }
+
+        cookie.remove('DATAPREP_WORKSPACE');
+
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setInitialized
+        });
+        this.eventEmitter.emit('DATAPREP_NO_WORKSPACE_ID');
+      });
+  }
+
+  componentWillUnmount() {
+    DataPrepStore.dispatch({
+      type: DataPrepActions.reset
+    });
+    this.eventEmitter.off('DATAPREP_BACKEND_DOWN', this.toggleBackendDown);
+  }
+
+  toggleBackendDown() {
+    this.setState({backendDown: true});
+  }
+
+  renderBackendDown() {
+    if (!this.state.backendDown) { return null; }
+
+    return (
+      <DataPrepServiceControl />
+    );
+  }
+
+  render() {
+    if (this.state.backendDown) { return this.renderBackendDown(); }
```

Comment:
I don't think this `if` check is necessary, since we're already doing that inside `renderBackendDown()`